gravitee-apim-cypress/cypress/integration/platform/jwt-policy-tests.ts

```diff
@@ -0,0 +1,320 @@
+      requestGateway(
+        {
+          url: `${Cypress.env('gatewayServer')}${rs256Api.context_path}`,
+          auth: { bearer: jwtToken_noSignature },
+        },
+        {
+          validWhen: (response) => response.status === 401,
+        },
+      )
+        .its('body')
+        .should('have.property', 'message')
+        .and('contain', 'Unauthorized');
+    });
+  });
+
+  describe('JWT Policy Authorization header propagation', () => {
+    it('should send Authorization header to backend API if auth header propagation is switched on', () => {
+      const jwtToken_hs256 = sign({ exp: 1900000000 }, secret, { noTimestamp: true });
+      requestGateway({
+        url: `${Cypress.env('gatewayServer')}${hs256Api.context_path}`,
+        auth: { bearer: jwtToken_hs256 },
+      }).should((response: Cypress.Response<any>) => {
+        expect(response.body.headers).to.have.property('Authorization', `Bearer ${jwtToken_hs256}`);
+      });
+    });
+
+    it('should not send Authorization header to backend API if auth header propagation is switched off', () => {
+      const jwtToken_hs256 = sign({ exp: 1900000000 }, secret, { noTimestamp: true });
+      requestGateway({
+        url: `${Cypress.env('gatewayServer')}${noAuthPropApi.context_path}`,
+        auth: { bearer: jwtToken_hs256 },
+      }).should((response: Cypress.Response<any>) => {
+        expect(response.body).to.have.property('query_params');
+        expect(response.body.headers).to.not.have.property('Authorization');
+      });
+    });
+  });
+
+  describe('JWT Policy using a JWKS endpoint', () => {
+    let jwksToken: string;
+    before(() => {
+      cy.log('-----  Retrieve access token from AM server  -----');
+      requestGateway({
+        method: 'POST',
+        url: `${Cypress.env('am_gatewayServer')}/auth/${am_domainHrid}/oauth/token`,
+        form: true,
+        auth: {
+          username: am_jwksTestApplication.settings.oauth.clientId,
+          password: am_jwksTestApplication.settings.oauth.clientSecret,
+        },
+        body: 'grant_type=client_credentials',
+      }).then((response: Cypress.Response<any>) => {
+        jwksToken = response.body.access_token;
+      });
+    });
+
+    after(() => {
+      am_deleteApplication(am_apiToken, am_domainId, am_jwksTestApplication.id);
+      am_deleteDomain(am_apiToken, am_domainId);
+    });
+
+    it('should successfully call API endpoint by using a JWKS URL (AM server) to verify access token', () => {
+      requestGateway({
+        url: `${Cypress.env('gatewayServer')}${jwksApi.context_path}`,
+        auth: { bearer: jwksToken },
+      }).should((response: Cypress.Response<any>) => {
+        expect(response.body).to.have.property('date');
+        expect(response.body).to.have.property('timestamp');
+      });
+    });
+
+    it('should fail to call API endpoint by using a JWKS URL (AM server) without access token', () => {
```

Comment:
Maybe we could also try to call it with an invalid accessToken ?

Reply:
That's right, I'll add that.